### DISCUSSION TITLE
how to rewrite ModbusSerialClient(method='rtu' , ...) for pymodbus 3.0.2 ?

### DISCUSSION THREAD

**Discussion by dreamflow** [pymodbus-dev/pymodbus#1177] [Help] [answered]
how should this pymodbus 2.5.3 code
```python
from pymodbus.client.sync import ModbusSerialClient`
client = ModbusSerialClient(method='rtu', port='COM31', baudrate=9600)
```
be rewritten for pymodbus 3.0.2 ?

instead of
`method='rtu'`
i've tried
`framer='rtu'`
`framer='ModbusRtuFramer'`
but got an error .

**Comment by janiversen** [comment, marked as answer]
Read the documentation

**Reply by dreamflow** [reply]
with pymodbus 3.0.2 i've found now via the documentation
```python
from pymodbus.transaction import ModbusRtuFramer
client = ModbusSerialClient(framer=ModbusRtuFramer, port='COM31', baudrate=9600)
```
so far so good 😊

but my code translation attempt of my next code line from pymodbus 2.5.3 syntax
```python
result = client.read_holding_registers(53, 2, unit=1)
```
 to pymodbus 3.0.2 syntax
```python
result = client.read_holding_registers(address=53, count=2, slave=1)
```
gives the
`NameError: name 'serial' is not defined`
what is not correct ?

**Reply by janiversen** [reply]
I have no idea ! it gives you a line and file name, which should help you identify where the problem is.

And btw, you syntax is valid, but this is not the most simple way to make the call, again look at the documentation.

Did you look at the examples, your import is also wrong, and you miss code.

**Reply by dhoomakethu** [reply]
You will have to install `pyserial` , it should have been installed by default. Please check if you are running inside correct python environement with all the required dependencies installed.

**Reply by dreamflow** [reply]
> You will have to install pyserial , it should have been installed by default.

that solved the problem ,
after installing pyserial it works fine 😃.
thank you @dhoomakethu and @janiversen .

i'm on python 3.11.0 on Win10-21H2 .
by running
`pip install pymodbus`
pyserial is not installed .
i have to also run
`pip install pyserial` .

**Comment by janiversen** [comment]
Pyserial is NOT installed by default, nor should is have been, see

https://github.com/riptideio/pymodbus#installing

@dhoomakethu I do not understand why you say it should be installed by default even in 2.5.3 pymodbus was pr default installed without dependencies.

**Reply by dhoomakethu** [reply]
OK, my mistake then.
